lambdas/matching/tests/matching-handler.test.ts

```diff
@@ -141,4 +177,82 @@ describe("matching-handler", () => {
       })
     );
   });
+
+  it("should concat all provided FamilyName", async () => {
+    const names = {
+      L: [
+        {
+          M: {
+            nameParts: {
+              L: [
+                {
+                  M: {
+                    type: {
+                      S: "GivenName",
+                    },
+                    value: {
+                      S: "John",
+                    },
+                  },
+                },
+                {
+                  M: {
+                    type: {
+                      S: "FamilyName",
+                    },
+                    value: {
+                      S: "Bob",
+                    },
+                  },
+                },
+                {
+                  M: {
+                    type: {
+                      S: "GivenName",
+                    },
+                    value: {
+                      S: "John",
+                    },
+                  },
+                },
+                {
+                  M: {
+                    type: {
+                      S: "FamilyName",
+                    },
+                    value: {
+                      S: "Alice",
+                    },
+                  },
+                },
+                {
+                  M: {
+                    type: {
+                      S: "GivenName",
+                    },
+                    value: {
+                      S: "John",
+                    },
+                  },
+                },
+                {
+                  M: {
+                    type: {
+                      S: "FamilyName",
+                    },
+                    value: {
+                      S: "Eve",
+                    },
+                  },
+                },
+              ],
+            },
+          },
+        },
+      ],
+    } as Names;
+
+    const result = extractSurname(names);
```

Comment:
test should be through the handler outputs

Reply:
Not possible because the handler returns the output of the API which in a unit test is the mocked return value. I can do this via the integration tests but this will require an update to the imposter mock.